--- a/Makefile	2017-11-24 22:22:19.000000000 +0000
+++ b/Makefile	2017-11-24 22:22:29.000000000 +0000
@@ -212,7 +212,7 @@
 # The task to run while instrument when building the profile-opt target
 # We exclude unittests with -x that take a rediculious amount of time to
 # run in the instrumented training build or do not provide much value.
-PROFILE_TASK=-m test.regrtest --pgo -x test_asyncore test_gdb test_multiprocessing test_subprocess
+PROFILE_TASK=-m test.regrtest -v --pgo -x test_asyncore test_gdb test_multiprocessing test_subprocess test_io test_logging test_posix
 
 # report files for gcov / lcov coverage report
 COVERAGE_INFO=	$(abs_builddir)/coverage.info
